### ISSUE TITLE
Opening two session after another results in undefined behaviour of user input

### ISSUE THREAD

**Issue by Nereboss** [varabyte/kotter#110]
**Describe the bug**

Opening a session after a previous session is closed results in undefined behaviour of user input.

**Dummy code for reproduction**

```
session {
    section { input() }.runUntilInputEntered()
}

session {
    section { input() }.runUntilInputEntered()
}
```

**To Reproduce**
1. Launch above dummy code
2. Input something for the first block (everything works)
3. Input something for the second block
4. It can be seen that some of the input will go through, other input will not show up and arrow keys produce weird characters when they go through ([D for left arrow)
5. Even for copied text only part of its characters dont register

**Expected behavior**
After a session is closed i expect a newly opened session behaves like the first session never existed

**Screenshots**
Which keystrokes get registered for the second block seems to be random, here is a screenshot of typing '1234567890' 3 times into both session blocks (a lot of keystrokes are missing for the 2nd one)
![image](https://github.com/varabyte/kotter/assets/105795397/29a6edcf-4939-4efa-9bc6-9f4c97f4d28e)

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version 1.1.1

**Additional context**
For small scripts of one file this could easily be avoided but to build more complex CLI tools where Kotter is used in multiple files this is very hard to get around without sacrificing code quality.

**Comment by Nereboss** [comment]
After further testing, this seems to be more a problem of the input method rather than sessions itself. I tried these three examples for testing:

Example 1:
```
session {
    section { text("test") }.run()
}

session {
    section { input() }.runUntilInputEntered()
}
```
Even though two sessions were started after another, input in the second session behaved like expected.

Example 2:
```
session {
    section { text("test") }.runUntilSignal { signal() }
}

session {
    section { input() }.runUntilInputEntered()
    section { input() }.runUntilInputEntered()
}
```
This example is closer to the one in the original issue but here everything works as expected as well.

Example 3:
```
session {
   section { input(isActive = false) }.run()
}

session {
    section { input() }.runUntilInputEntered()
}
```
This example opens two inputs in two different sections but the 2nd input still behaves normal.

My guess would be that when opening a new session or during closing of the previous session, the first input is not deactivated, resulting in both being active at the same time.

**Comment by Nereboss** [comment]
@bitspittle any idea why this issue might occur? It also seems to produce different behaviour on unix and windows machines and also behave differently in powershell and git bash

**Comment by bitspittle** [comment]
Huge apologies for not replying! I just had an insanely busy week. Literally just started coming up for air today.

It definitely sounds like a bug.

Note that different OSes have different terminal implementations.

Unix: https://github.com/varabyte/kotter/blob/main/kotter/src/posixMain/kotlin/com/varabyte/kotter/terminal/native/NativeTerminal.kt

Windows:
https://github.com/varabyte/kotter/blob/main/kotter/src/winMain/kotlin/com/varabyte/kotter/terminal/native/NativeTerminal.kt

Although I should check -- are you using the JVM target when you build your app? Or the specific native targets?

If you're targeting the JVM, I'm using a library for that (JANSI?), and they too have different Windows and Unux implementations.

I assume git bash on Windows is actually using a Unix emulator, but I'm not 100% sure. Still, if so, that could make the difference make sense.

**Comment by Nereboss** [comment]
We only use the JVM target. 
We are currently trying to overhaul the cli of another open source project ([CodeCharta](https://github.com/MaibornWolff/codecharta)). 
Just linking it here in case there are any project specific questions that can be answered with a quick look at the project^^

**Comment by bitspittle** [comment]
@Nereboss I know it's been quiet here, so first I wanted to say thanks so much for the report.  I just started looking into this today (sorry, I was super stuck in Kobweb stuff for the last two weeks).

I'm able to repro, and the issue looks very much like it's in my code.

If all goes well, I'll have a 1.1.2-SNAPSHOT build up by end of day you can test.

**Comment by bitspittle** [comment]
OK, I've published a new snapshot. @Nereboss if you want to test this, please update Kotter to 1.1.2-SNAPSHOT in your libs.versions.toml file.

You also need to add a special maven snapshots repository, which I just added a new README section about here: https://github.com/varabyte/kotter/?tab=readme-ov-file#testing-snapshots

**Comment by Nereboss** [comment]
Tested it out and everything works as expected on mac, windows and in git bash 👍

**Comment by bitspittle** [comment]
Thanks for the confirmation.

I'm aiming to release a new Kotter version in the next day or two. You can continue to use the snapshot version if you'd like -- things are not going to change appreciably between that and the final version (unless you found yourself using [the new `grid` builder](https://github.com/varabyte/kotter/tree/1.1.2?tab=readme-ov-file#grids), at which point there's still a bit of work landing today)

**Comment by bitspittle** [comment]
This should now be fixed in 1.1.2. Thanks for the report!
